src/PockyBot.NET/Services/Triggers/StringConfig.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using GlobalX.ChatBots.Core.Messages;
+using PockyBot.NET.Constants;
+using PockyBot.NET.Persistence.Repositories;
+
+namespace PockyBot.NET.Services.Triggers
+{
+    internal class StringConfig : ITrigger
+    {
+        private readonly IConfigRepository _configRepository;
+        public string Command => Commands.StringConfig;
+        public bool DirectMessageAllowed => false;
+        public bool CanHaveArgs => true;
+        public string[] Permissions => new[] {Roles.Admin, Roles.Config};
+
+        public StringConfig(IConfigRepository configRepository)
+        {
+            _configRepository = configRepository;
+        }
+
+        public async Task<Message> Respond(Message message)
+        {
+            var command = string.Join("", message.MessageParts.Skip(1).Select(x => x.Text)).Trim().Remove(0, 4).Trim();
+            var commandWords = command.Split(' ').Where(x => !string.IsNullOrWhiteSpace(x)).ToList();
+
+            if (commandWords.Count() < 2)
+            {
+                return new Message
+                {
+                    Text = $"Please specify a command. Possible values are {string.Join(", ", ConfigActions.All())}."
+                };
+            }
+
+            string newMessage;
+            switch (commandWords[1])
+            {
+                case ConfigActions.Get:
+                    newMessage = GetStringConfig();
+                    break;
+                case ConfigActions.Add:
+                    newMessage = await AddStringConfig(commandWords).ConfigureAwait(false);
+                    break;
+                case ConfigActions.Delete:
+                    newMessage = await DeleteStringConfig(commandWords).ConfigureAwait(false);
+                    break;
+                default:
+                    newMessage = $"Invalid string config command. Possible values are {string.Join(", ", ConfigActions.All())}.";
+                    break;
+            }
+
+            return new Message
+            {
+                Text = newMessage
+            };
+        }
+
+        private string GetStringConfig()
+        {
+            var stringConfig = _configRepository.GetAllStringConfig();
+            var groupedConfig = stringConfig.GroupBy(x => x.Name);
+
+            var stringBuilder = new StringBuilder();
```

Comment:
Minor: this can be initialised with a string, being the initial value (so line 68)